### PR TITLE
Doc: Add FAQ to speed up parsing with tons of dag files

### LINKED ISSUE #17437: It's too slow to recognize new dag file when there are a log of dags files
**Description**
There are 5000+ dag files in our production env. It will delay for almost 10mins to recognize the new dag file if a new one comes. 

There are 16 CPU cores in the scheduler machine. The airflow version is 2.1.0.

**Use case / motivation**

I think there should be a feature to support recognizing new dag files or recently modified dag files faster.

**Are you willing to submit a PR?**
Maybe will.

**Related Issues**

### PR DESCRIPTION
**Pull request by kaxil**
This feature was added in https://github.com/apache/airflow/pull/16075. This PR adds it to docs to avoid situations like https://github.com/apache/airflow/issues/17437

closes https://github.com/apache/airflow/issues/17437

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by jedcunningham** [comment]
Might be worth mentioning this only helps with "simple" DAGs, e.g. if the main DAG file itself actually changes. Not sure exactly how to phrase it in this context, but this won't always be a safe option.

File: docs/apache-airflow/faq.rst [hunk 1]
```diff
@@ -119,6 +119,18 @@ How do I trigger tasks based on another task's failure?
 
 You can achieve this with :ref:`concepts:trigger-rules`.
 
+When there are a lot of dags files, how to speed up parsing of new files?
```
Comment (eladkal) [inline docs/apache-airflow/faq.rst]: Should we be more specific to what is a lot?

Comment (kaxil) [reply docs/apache-airflow/faq.rst]: Updated

**Comment by kaxil** [comment]
> Might be worth mentioning this only helps with "simple" DAGs, e.g. if the main DAG file itself actually changes. Not sure exactly how to phrase it in this context, but this won't always be a safe option.

Updated

**Review by eladkal — APPROVED** [review]

**Pull request merged**